app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import androidx.navigation.NavController
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.components.browser.state.selector.getNormalOrPrivateTabs
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.ShareData
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.home.HomeFragment
+
+/**
+ * For interacting with UI that is specifically for [TabsTrayFragment[]] and other browser
+ * tab tray views.
+ */
+interface NavigationInteractor {
+
+    /**
+     * Called when tab tray should be dismissed.
+     */
+    fun onTabTrayDismissed()
+
+    /**
+     * Called when user clicks the share tabs button.
+     */
+    fun onShareTabsOfTypeClicked(private: Boolean)
+
+    /**
+     * Called when user clicks the tab settings button.
+     */
+    fun onTabSettingsClicked()
+
+    /**
+     * Called when user clicks the close all tabs button.
+     */
+    fun onCloseAllTabsClicked(private: Boolean)
+
+    /**
+     * Called when user clicks the recently closed tabs menu button.
+     */
+    fun onOpenRecentlyClosedClicked()
+}
+
+/**
+ * A default implementation of [NavigationInteractor].
+ */
+class DefaultNavigationInteractor(
```

Comment:
Looks great! I don't see this code evolving much as we iterate. If you feel the same, I think we can add tests with some simple assertions.

Reply:
Sure, I'll add the tests